### DISCUSSION TITLE
Use of HTML native  <dialog> element for modals, offcanvas, popovers, & tooltips

### DISCUSSION THREAD

**Discussion by JSystemsTech** [twbs/bootstrap#38560] [Ideas] [answered]
I've been reading recently that most (if not all) commonly used browsers now support use of the native html dialog element, which is used for custom modal and popover like creation without bootstrap. 

My question is are there any plans for  bootstrap to use this element at some point to create any of the following components: modals, offcanvases, popovers, & tooltips? 

some benefits to using this are that accessibility is mostly built in and interpreted by the browser, backdrop is included natively (can be styled with CSS), and escape keys automatically close a modal instances just to name a few.  

I've been reading over this dev's blog and he seems to have a solid understanding of the element works and goes into more detail.

[Modals Will Never Be The Same - HTML dialog Element](https://blog.webdevsimplified.com/2023-04/html-dialog/)

below is a basic snippet

### HTML
```html
<dialog open>
  <span>You can see me</span>
</dialog>
```

### JS
```javascript
const dialog = document.querySelector("dialog");
dialog.show(); // Opens a non-modal dialog
dialog.showModal(); // Opens a modal
```

**Comment by julien-deramond** [comment, marked as answer]
> My question is are there any plans for bootstrap to use this element at some point to create any of the following components: modals, offcanvases, popovers, & tooltips?

Yep, it's tracked in https://github.com/twbs/bootstrap/issues/33804 for modals first. Hopefully in v6 depending on other prioritized features and the upgrade of our `.browserslistrc` (and so the browsers' support).

**Reply by JSystemsTech** [reply]
Awesome! Looking forward to seeing how that ends up getting integrated! The project I work with has had to implement a custom modal component using dialog to fulfill accessibility requirements so this is great news to know that it is an upcoming feature.
